### PR TITLE
provider/aws: New Resource aws_codedeploy_deployment_config

### LINKED ISSUE #10889: suggestion new resource aws_codedeploy_deployment_configuration
Suggestion to add new resource to manage AWS CodeDeploy deployment configuration.

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_codedeploy_deployment_configuration

### Terraform Configuration Files
```hcl
resource "aws_codedeploy_deployment_configuration" "foo" {
        minimum_healthy_hosts {
            type = "HOST_COUNT" | "FLEET_PERCENT:"
            value = 0
        }
}
```

Reference to AWS CLI help on this resource:
http://docs.aws.amazon.com/cli/latest/reference/deploy/create-deployment-config.html

### PR DESCRIPTION
**Pull request by stack72**
Fixes #10889 

```
% make testacc TEST=./builtin/providers/aws TESTARGS='-run=TestAccAWSCodeDeployDeploymentConfig_'                       2 ↵ ✹ ✭
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2017/01/05 20:49:34 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/aws -v -run=TestAccAWSCodeDeployDeploymentConfig_ -timeout 120m
=== RUN   TestAccAWSCodeDeployDeploymentConfig_fleetPercent
--- PASS: TestAccAWSCodeDeployDeploymentConfig_fleetPercent (21.17s)
=== RUN   TestAccAWSCodeDeployDeploymentConfig_hostCount
--- PASS: TestAccAWSCodeDeployDeploymentConfig_hostCount (19.32s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	40.508s
```

### CODE REVIEW

File: builtin/providers/aws/resource_aws_codedeploy_deployment_config.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		DeploymentConfigName: aws.String(d.Get("deployment_config_name").(string)),
+		MinimumHealthyHosts:  expandAwsCodeDeployConfigMinimumHealthHosts(d),
+	}
+
+	_, err := conn.CreateDeploymentConfig(input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("deployment_config_name").(string))
```
Comment (paddyforan) [inline builtin/providers/aws/resource_aws_codedeploy_deployment_config.go]: Should we be using the name as the ID if it has an ID property?

File: website/source/docs/providers/aws/r/codedeploy_deployment_config.html.markdown [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Example Usage
+
+```
+
+resource "aws_codedeploy_deployment_config" "foo" {
+    deployment_config_name = "test-deployment-config"
+    minimum_healthy_hosts {
+        type = "HOST_COUNT"
+        value = "2"
```
Comment (paddyforan) [inline website/source/docs/providers/aws/r/codedeploy_deployment_config.html.markdown]: Should this be quoted?

Comment (stack72) [inline website/source/docs/providers/aws/r/codedeploy_deployment_config.html.markdown]: This is an int - no need to quote this :)

File: builtin/providers/aws/resource_aws_codedeploy_deployment_config.go (hunk 1 above)
Comment (stack72) [inline builtin/providers/aws/resource_aws_codedeploy_deployment_config.go]: I used the name as the Terraform ID here as this is what we refer to when calling from other resources

i.e. deployment_config_name = "${aws_codedeploy_deployment_config.foo.id}"

Comment (paddyforan) [inline builtin/providers/aws/resource_aws_codedeploy_deployment_config.go]: Fair.

File: website/source/docs/providers/aws/r/codedeploy_deployment_config.html.markdown (hunk 2 above)
Comment (stack72) [inline website/source/docs/providers/aws/r/codedeploy_deployment_config.html.markdown]: d'oh! removing now

**Comment by paddyforan** [comment]
LGTM on green travis.

**Pull request merged**